app/containers/MessageBox/index.js

```diff
@@ -181,14 +190,23 @@ class MessageBox extends Component {
 	// eslint-disable-next-line react/sort-comp
 	debouncedOnChangeText = debounce((text) => {
 		this.setInput(text);
+		const slashCommand = text.match(/^\/([a-z0-9._-]+) (.+)/im);
+		if (slashCommand) {
+			const [, name, params] = slashCommand;
+			const command = database.objects('slashCommand').filtered('command == $0', name);
+			if (command && command[0] && command[0].providesPreview) {
+				return this.setCommandPreview(name, params);
+			}
+		}
 
 		if (this.component) {
 			requestAnimationFrame(() => {
 				const { start, end } = this.component._lastNativeSelection;
 				const cursor = Math.max(start, end);
 				const lastNativeText = this.component._lastNativeText;
-				const regexp = /(#|@|:)([a-z0-9._-]+)$/im;
+				const regexp = /(#|@|:|^\/)([a-z0-9._-]+)$/im;
 				const result = lastNativeText.substr(0, cursor).match(regexp);
+				this.showCommandPreview = false;
```

Comment:
why `showCommandPreview` is not in state? 🤔 and this would fit into a function `hide\showCommandPreview`. What do you think?

Reply:
I don't want  to rerender  by that , so it can be a constant ..